### ISSUE TITLE
Spikes element does not overlay properly

### ISSUE THREAD

**Issue by jlstevens** [ioam/holoviews#1539]
As shown here:

![image](https://user-images.githubusercontent.com/890576/27157241-8d9b7ac4-5158-11e7-9865-4b83bb67c27f.png)

I get the same issue with matplotlib.

**Comment by jlstevens** [comment]
Actually, this issue may be a duplicate of the issue described in #1480.

**Comment by jlstevens** [comment]
Looks like it is fixed?

![image](https://user-images.githubusercontent.com/890576/27766780-694b6362-5ed4-11e7-8735-5f857907c997.png)

This screenshot shows a different issue (unexpected colormapping) described in #1669.

**Comment by philippjfr** [comment]
I'm not sure this was ever a real issue, it's just to do with the semantics of Spikes. By default ``Spikes`` do not have a height so you have to set the height as a plot option. In the plot above the height of the Spikes is likely just too small to see. In your new example you have given them a height so there is no issue.

**Comment by jlstevens** [comment]
Philipp and I discussed this about maybe the height plot options should be 1 by default as a relative quantity (fraction of the y-axis). That would make more sense to me I think.

**Comment by neighthan** [comment]
@jlstevens I agree. I just learned that `VLine` can't appear in the legend, so I replaced it with `Spikes` in my code. However, it's not easy to get spikes that overlay properly when I want them to be the full height of the plot. Now I have to modify my spikes based on what I'm overlaying them with, which isn't nice. Unless explicit heights are given, I think having full-height spikes is the most reasonable default.

**Comment by SandervandenOord** [comment]
> Philipp and I discussed this about maybe the height plot options should be 1 by default as a relative quantity (fraction of the y-axis). That would make more sense to me I think.

Is there already something like relative spike_length, so you don't have to think yourself about what absolute length the spikes should get?
Right now I would do something like ```scatter * spikes.opts(spike_length='very large number')``` to be sure it goes over the whole y range of scatter.
But yeah what do you do when you want the spikes to be at 25% of the y-range?

**Comment by jlstevens** [comment]
@philippjfr I'm wondering if there is a way to use the `dim` object to find the y-range and scale as a proportion of that. Should be possible, right?

**Comment by philippjfr** [comment]
I'm not sure if `spike_length` accepts a dim transform atm but in theory that should be possible yes. Something like:

```python
spike_length=np.ptp(dim('y'))/10.
```

**Comment by philippjfr** [comment]
Actually no that doesn't make sense, that would just get the range of the y-values, not the overall range of the plot.
